### PR TITLE
dumpling: supports foreign key limitations

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/39913

Problem Summary:

### What is changed and how it works?
Add specical comments for dumpling exported sqls.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test

Side effects

- None

Documentation

- None

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: dumpling/export/ir_impl.go [hunk 1]
```diff
@@ -370,3 +371,22 @@ func (td *multiQueriesChunk) Close() error {
 func (*multiQueriesChunk) RawRows() *sql.Rows {
 	return nil
 }
+
+var serverSpecialComments = map[version.ServerType][]string{
```
Comment (gozssky) [inline dumpling/export/ir_impl.go:375]: Are these comments for dump files or queries?

Comment (lance6716) [reply dumpling/export/ir_impl.go:375]: seems they are written to dump files. These comments take effects when importing not dumping, so we should not using upstream database to decide comments

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: dumpling/export/ir_impl.go (hunk 1 above)
Comment (lichunzhu) [reply dumpling/export/ir_impl.go:375]: These comments are used for dumped data files. They are used for scenes that we directly restore sqls back through MySQL cmd.
For import tools, they need to re-decide the used comments. These changes are copied from mydumper.

Comment (lance6716) [reply dumpling/export/ir_impl.go:375]: need an integration test with lightning later to check we can handle mariadb case😂

Comment (lance6716) [reply dumpling/export/ir_impl.go:375]: https://github.com/pingcap/tiflow/blob/099da1e08f1310c2366093f2c314a51e0cde772b/dm/loader/loader.go#L1218

DM loader may not be able to handle `SET FOREIGN_KEY_CHECKS=0;`, not sure what's the behaviour for lightning

**Review by lance6716 — APPROVED** [review]

File: dumpling/export/ir_impl.go (hunk 1 above)
Comment (lichunzhu) [reply dumpling/export/ir_impl.go:375]: I think Lightning shouldn't read these comments😂 It should add its own comments according to downstream database.

**Review by gozssky — APPROVED** [review]

File: dumpling/export/ir_impl.go (hunk 1 above)
Comment (lichunzhu) [reply dumpling/export/ir_impl.go:375]: > 

I think DM loader's handle is correct. Loader should execute foreign key handlement itself when it finds its downstream is TiDB. See https://github.com/mydumper/mydumper/blob/e4d05b1a76b7a529d24c43a4d043790f355c5085/src/myloader.c#L388

**Pull request merged**